src/profile/ProfileForm.tsx

```diff
@@ -37,585 +23,315 @@ interface ChangedUser {
-            onChange={onChange}
-            isValid={"first_name" in state.changedUser}
+          <Field
+            errors={props.context.errors}
+            type="email"
+            name="email"
+            ref={props.context.register({
+              required: "USER.REQUIRED.EMAIL",
+              maxLength: {
+                value: 254,
+                message: "USER.EMAIL.TOO_LONG.254",
+              },
+              pattern: {
+                value: /\S+@\S+\.\S+/,
+                message: "USER.EMAIL.INVALID",
+              },
+            })}
           />
         </Col>
       </Form.Group>
-
-      <Form.Group as={Row} controlId="profileWriterName">
+      <Form.Group as={Row} controlId="writerName">
         <Form.Label column sm={2}>
-          Writer name
+          Writer Name:
         </Form.Label>
         <Col sm={10}>
-          <Form.Control
+          <Field
+            errors={props.context.errors}
+            type="text"
             name="writer_name"
-            placeholder="e.g. Grower of Apples"
-            value={state.changedUser.writer_name ?? user?.writer_name ?? ""}
-            onChange={onChange}
-            isValid={(state.changedUser.writer_name?.length ?? 0) > 0}
-            isInvalid={
-              state.errors.writer_name && state.errors.writer_name.length > 0
-            }
+            ref={props.context.register}
           />
-          <Form.Control.Feedback type="invalid">
-            <ul>
-              {state.errors.writer_name?.map((msg) => (
-                <li key={msg}>{ERRORS[msg]}</li>
-              ))}
-            </ul>
-          </Form.Control.Feedback>
         </Col>
       </Form.Group>
-
-      {auth.isEditor() && (
-        <Form.Group as={Row} controlId="profileIsEditor">
+      {auth.user?.is_editor && (
+        <Form.Group as={Row} controlId="isEditor">
           <Form.Label column sm={2}>
-            Role
+            Status:
           </Form.Label>
           <Col sm={10}>
-            {/* NOTE: The custom attribute hasn't landed yet; it's here for future-compat. In the meantime, we add a
-              custom class instead. */}
             <Form.Control
               as="select"
-              name="is_editor"
-              onChange={onChange}
-              value={state.changedUser.is_editor?.toString()}
-              disabled={
-                auth.user?.username === user?.username && !user?.is_staff
+              onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
+                if (e.currentTarget.value === "editor") {
+                  props.context.setValue("is_editor", true);
+                } else if (e.currentTarget.value === "contributor") {
+                  props.context.setValue("is_editor", false);
+                }
+              }}
+              value={
+                props.context.getValues().is_editor ? "editor" : "contributor"
               }
-              className="custom-select"
-              custom
```

Comment:
Missing these two attributes for custom styling.

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/13/commits/53ece12a397300e594d32c8dc8ce2d7c9859a9ed